lightning/src/routing/router.rs

```diff
@@ -1794,11 +1870,20 @@ where L::Target: Logger {
 		// If a caller provided us with last hops, add them to routing targets. Since this happens
 		// earlier than general path finding, they will be somewhat prioritized, although currently
 		// it matters only if the fees are exactly the same.
-		let route_hints = match &payment_params.payee {
-			Payee::Clear { route_hints, .. } => route_hints,
-			_ => return Err(LightningError{err: "Routing to blinded paths isn't supported yet".to_owned(), action: ErrorAction::IgnoreError}),
-		};
-		for route in route_hints.iter().filter(|route| !route.0.is_empty()) {
+		for hint in blinded_route_hints.iter() {
+			let have_hop_src_in_graph =
+				// Only add the hops in this route to our candidate set if either
+				// we have a direct channel to the first hop or the first hop is
+				// in the regular network graph.
+				first_hop_targets.get(&NodeId::from_pubkey(&hint.src_node_id)).is_some() ||
+				network_nodes.get(&NodeId::from_pubkey(&hint.src_node_id)).is_some();
+			if have_hop_src_in_graph {
+				add_entry!(CandidateRouteHop::PrivateHop { hint },
```

Comment:
I'm not 100% sure about this, because the clear route hints case does something different if our first hop channels are connected to our last hop route hints.